src/opentime/timeRange.h

```diff
@@ -82,6 +82,34 @@ class TimeRange {
         return _start_time < other.end_time_exclusive() && other._start_time < end_time_exclusive();
     }
 
+    bool before(TimeRange other) const {
+        return _start_time < other._start_time && end_time_exclusive() <= other._start_time;
+    }
+
+    bool before(RationalTime other) const {
+        return end_time_exclusive() < other;
+    }
+
+    bool meets(TimeRange other) const {
+        return end_time_exclusive() == other._start_time || _start_time == other.end_time_exclusive();
+    }
+
+    bool begins(TimeRange other) const {
+        return _start_time == other._start_time;
+    }
+
+    bool begins(RationalTime other) const {
+        return _start_time == other;
+    }
+
+    bool finishes(TimeRange other) const {
+        return end_time_exclusive() == other.end_time_exclusive();
+    }
+
+    bool finishes(RationalTime other) const {
+        return end_time_exclusive() == other;
```

Comment:
We are asking for end time exclusive in all cases for these comparisons. We should state, in the documentation what the extended rules for Allen's Algebra are for clusivity. For example, if both ranges are inclusive at both ends, and the operation is "meets" we have to decide if that is "meet" or an "overlap". I would request that a prerequisite for this part of the project is to write up a table with all the operators, versus end point clusivity combinations. Also, the associated unit test should exercise the complete table to validate that the operations conform to the clusivity described in the table.